lib/dataset/include/scipp/dataset/bins_reduction.h

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Jan-Lukas Wynen
+#pragma once
+
+#include "scipp/variable/variable.h"
+
+#include "scipp-dataset_export.h"
+
+// These functions are placed in scipp::variable for ADL but cannot be easily
+// moved into that compilation module since the implementation requires
+// DataArray.
```

Comment:
Does this still apply? It looks like it could be moved now?

Reply:
You are right. They can be moved because of the new function in `VariableFactory`.